presto-native-execution/presto_cpp/main/operators/FileSystemBroadcast.cpp

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "presto_cpp/main/operators/FileSystemBroadcast.h"
+#include "velox/serializers/PrestoSerializer.h"
+using namespace facebook::velox::exec;
+using namespace facebook::velox;
+
+namespace facebook::presto::operators {
+
+FileSystemBroadcast::FileSystemBroadcast(const std::string& rootPath)
+    : rootPath_(rootPath) {
+  // VELOX_CHECK_NOT_NULL(rootPath, "Root path for broadcast files is null!");
+  fileSystem_ = velox::filesystems::getFileSystem(rootPath, nullptr);
+}
+
+std::shared_ptr<BroadcastFileReader> FileSystemBroadcast::createReader(
+    velox::memory::MemoryPool* FOLLY_NONNULL pool) {
+  return nullptr;
+}
+
+std::shared_ptr<BroadcastFileWriter> FileSystemBroadcast::createWriter(
+    velox::memory::MemoryPool* FOLLY_NONNULL pool) {
+  // auto filename = getFileName(rootPath_, taskId, StageId);
+  fileSystem_->mkdir(rootPath_);
+  auto filename = fmt::format("{}/file_broadcast.bin", rootPath_);
+  LOG(INFO) << "Opening file for broadcast: " << filename;
+  auto writeFile = fileSystem_->openFileForWrite(filename);
+  auto broadcastFileWriter = std::make_shared<BroadcastFileWriter>(
+      std::move(writeFile), filename, pool);
+  return broadcastFileWriter;
+}
+
+BroadcastFileWriter::BroadcastFileWriter(
+    std::unique_ptr<WriteFile> writeFile,
+    std::string_view filename,
+    velox::memory::MemoryPool* FOLLY_NONNULL pool)
+    : writeFile_(std::move(writeFile)), filename_(filename), pool_(pool) {}
+
+void BroadcastFileWriter::collect(RowVectorPtr input) {
+  serialize(input);
+}
+void BroadcastFileWriter::noMoreData() {
+  writeFile_->flush();
+}
+RowVectorPtr BroadcastFileWriter::fileStats() {
+  auto data = BaseVector::create<FlatVector<StringView>>(VARCHAR(), 1, pool_);
+  data->set(0, filename_);
+  return std::make_shared<RowVector>(
+      pool_,
+      ROW({"filepath"}, {VARCHAR()}),
+      BufferPtr(nullptr),
+      1,
+      std::vector<VectorPtr>({std::move(data)}));
+}
+
+void BroadcastFileWriter::serialize(
+    const RowVectorPtr& rowVector,
+    const VectorSerde::Options* serdeOptions) {
+  auto numRows = rowVector->size();
+  std::vector<IndexRange> rows(numRows);
+  auto arena = std::make_unique<StreamArena>(pool_);
+  auto rowType = asRowType(rowVector->type());
+  auto serde_ = std::make_unique<serializer::presto::PrestoVectorSerde>();
```

Comment:
serde_ shall be created on ctor? rowType shall also be specified in BroadcastFileWriter ctor?